{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "80098b2b",
   "metadata": {},
   "source": [
    "SIMULATED DATA:\n",
    "Simulated reads were generated using Escherichia coli BL21 reference genome.\n",
    "Short-read simulation done via ART.\n",
    "Long-read simulation done via badread.\n",
    "Each experiment was repeated three times for three replicates.\n",
    "Short-read mean fragment length always set to 200bp in length.\n",
    "Long-read mean fragment length always set to 7000bp, 400bp stdev. (See experiment 3).\n",
    "All assembled at 32 cores."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7a396e44",
   "metadata": {},
   "source": [
    "EXPERIMENT 1:\n",
    "Simulating read quality (accuracy) at varying levels (high, low, and mid). \n",
    "Short-reads generated at MSv3 (high quality), HS20 (moderate quality), and GAIIx (poor quality), respectively. \n",
    "Long-reads generated at (identity%, stdev) 95,2.5 , 73,10 , and 50,15 respectively.\n",
    "Short-read coverage always set to 50x.\n",
    "Long-read coverage always set to 100x."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "73325ad8",
   "metadata": {},
   "source": [
    "HIGH:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "62d30be3",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7da188d0",
   "metadata": {},
   "source": [
    "-ss Quality Model, -i reference genome, -p paired end reads, -l length, -f coverage, -m mean average length, -s stdev, -o output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2057fa33",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss MSv3 -i referenceg/BL21.fasta -p -l 200 -f 50 -m 300 -s 3 -o bench_proj/ART/exp1high/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46522822",
   "metadata": {},
   "source": [
    "Long-reads (split into multiple jobs to reduce CPU load and consequently runtime): "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f87daca2",
   "metadata": {},
   "source": [
    "--reference reference genome, --quantity coverage, --length read length --error_model error model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cd126db",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > h.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > i.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > j.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq i.fastq j.fastq > bench_proj/badread/exp1high/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35f81a9e",
   "metadata": {},
   "source": [
    "MID:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b73bea9b",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef1b60ea",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss HS20 -i referenceg/BL21.fasta -p -l 200 -f 50 -m 300 -s 3 -o bench_proj/ART/exp1mid/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f6a47f3",
   "metadata": {},
   "source": [
    "Long-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ba04a91",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > h.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > i.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 73,10 --length 7000,400 --error_model nanopore2023 > j.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq i.fastq j.fastq > bench_proj/badread/exp1mid/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b70c35aa",
   "metadata": {},
   "source": [
    "LOW:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e863b170",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f3a78aa",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss GA2 -i referenceg/BL21.fasta -p -l 100 -f 50 -m 300 -s 3 -o bench_proj/ART/exp1low/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "00b6a1ed",
   "metadata": {},
   "source": [
    "Long-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "55cd922b",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > h.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > i.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --identity 50,15 --length 7000,400 --error_model nanopore2023 > j.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq i.fastq j.fastq > bench_proj/badread/exp1low/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a19f8646",
   "metadata": {},
   "source": [
    "ASSEMBLY GENERATION:\n",
    "Each assembly was generated using each respective quality tier. Ex: Tool assembled using exp1mid short reads replicate 3, and exp1mid long read replicate 3. \n",
    "NOTE: Each tool only shows one replicate generation to save space. Assemblies were generated for each tier (High, Mid, or Low) and for each replicate per tier (1, 2, 3). Only folder location differs in code."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74296b2b",
   "metadata": {},
   "source": [
    "UNICYCLER:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d5df02f2",
   "metadata": {},
   "source": [
    "--verbose time and space complexity -1 first short read -2 second short read -l long read -t threads -o output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15f29168",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose unicycler -1 bench_proj/ART/exp1/rep1_1.fq -2 bench_proj/ART/exp1high/rep1_2.fq -l bench_proj/badread/exp1high/rep1.fastq -t 32 -o bench_proj/unicycler/exp1high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88dd67e1",
   "metadata": {},
   "source": [
    "FLYE:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7815bb37",
   "metadata": {},
   "source": [
    "--nano-raw read type and file specification --genome-size size of genome --out-dir output filepath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b529e18",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose flye --nano-raw bench_proj/badread/exp1high/rep1.fastq --genome-size 4.7m --out-dir bench_proj/flye/exp1high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "448e42b7",
   "metadata": {},
   "source": [
    "SPADES"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "20405033",
   "metadata": {},
   "source": [
    "-1 first short read -2 second short read --nanopore long read type and file -t threads -o output file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dcbd42d",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose spades.py -1 bench_proj/ART/exp1high/rep1_1.fq -2 bench_proj/ART/exp1high/rep1_2.fq --nanopore bench_proj/badread/exp1high/rep1.fastq -t 32 -o bench_proj/spades/exp1high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6430016d",
   "metadata": {},
   "source": [
    "EXPERIMENT 2:\n",
    "Simulating read coverage at varying levels (high, mid, low). \n",
    "Short-reads generated at 70x, 35x, and 17x, respectively. \n",
    "Long-reads generated at 100x, 50x, and 15x, respectively.\n",
    "Short-reads always set to MSv3.\n",
    "Long-reads always set to 95, 2.5 (%identity, stdev)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4901418",
   "metadata": {},
   "source": [
    "HIGH:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "89f3c753",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "13403ba8",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss MSv3 -i referenceg/BL21.fasta -p -l 200 -f 70 -m 300 -s 3 -o bench_proj/ART/exp2high/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acd0c095",
   "metadata": {},
   "source": [
    "Long-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56621f6c",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > h.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > i.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > j.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq i.fastq j.fastq > bench_proj/badread/exp2high/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8fbd6291",
   "metadata": {},
   "source": [
    "MID:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "34f5e1d2",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0bd780b",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss MSv3 -i referenceg/BL21.fasta -p -l 200 -f 35 -m 300 -s 3 -o bench_proj/ART/exp2mid/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9ce6e82b",
   "metadata": {},
   "source": [
    "Long-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f52f4b44",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > h.fastq badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > i.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > j.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq i.fastq j.fastq > bench_proj/badread/exp2mid/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c3aef8e4",
   "metadata": {},
   "source": [
    "LOW:"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0545445e",
   "metadata": {},
   "source": [
    "Short-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "238a076e",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "art_illumina -ss MSv3 -i referenceg/BL21.fasta -p -l 100 -f 17 -m 300 -s 3 -o bench_proj/ART/exp2low/rep1_"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "067efac2",
   "metadata": {},
   "source": [
    "Long-reads:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57602ae6",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 10x --length 7000,400 --error_model nanopore2023 > e.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq > bench_proj/badread/exp2low/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f742b2d",
   "metadata": {},
   "source": [
    "ASSEMBLY GENERATION:\n",
    "Each assembly was generated using each respective quality tier. Ex: Tool assembled using exp2mid short reads replicate 3, and exp2mid long read replicate 3. \n",
    "NOTE: Each tool only shows one replicate generation to save space. Assemblies were generated for each tier (High, Mid, or Low) and for each replicate per tier (1, 2, 3). Only folder location differs in code."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0e16a5f",
   "metadata": {},
   "source": [
    "UNICYCLER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2e974da",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose unicycler -1 bench_proj/ART/exp2/rep1_1.fq -2 bench_proj/ART/exp2high/rep1_2.fq -l bench_proj/badread/exp2high/rep1.fastq -t 32 -o bench_proj/unicycler/exp21high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48abd3a4",
   "metadata": {},
   "source": [
    "FLYE:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "15f8ae59",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose flye --nano-raw bench_proj/badread/exp2high/rep1.fastq --genome-size 4.7m --out-dir bench_proj/flye/exp2high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97059845",
   "metadata": {},
   "source": [
    "SPADES:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "24c928b3",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose spades.py -1 bench_proj/ART/exp2high/rep1_1.fq -2 bench_proj/ART/exp2high/rep1_2.fq --nanopore bench_proj/badread/exp2high/rep1.fastq -t 32 -o bench_proj/spades/exp2high/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33dab18c",
   "metadata": {},
   "source": [
    "EXPERIMENT 3:\n",
    "Simulating Oxford-Nanopore long reads in 2018 and 2024 based on publications around those times. Testing the effect of long-read quality improving over time.\n",
    "2018 long reads were simulated at 85% identity and 9600 average read length (reads were longer and less accurate) with a coverage of 30x.\n",
    "2024 long reads were simulated at 99% identity and 6000 average read length with a coverage of 40x. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7715014a",
   "metadata": {},
   "source": [
    "2018:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69e1df74",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5 --length 9600,600 --error_model nanopore2018 > a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5  --length 9600,600 --error_model nanopore2018 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5 --length 9600,600 --error_model nanopore2018 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5 --length 9600,600 --error_model nanopore2018 > d.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5 --length 9600,600 --error_model nanopore2018 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 85,5 --length 9600,600 --error_model nanopore2018 > f.fastq & wait & cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq > bench_proj/badread/exp3/2018/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "24293a2f",
   "metadata": {},
   "source": [
    "2024:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2282698b",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023> a.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1  --length 6000,1200 --error_model nanopore2023 > b.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > c.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > d.fastq & badread simulate â€“reference/referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > e.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > f.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > g.fastq & badread simulate --reference referenceg/BL21.fasta --quantity 5x --identity 99,1 --length 6000,1200 --error_model nanopore2023 > h.fastq & wait && cat a.fastq b.fastq c.fastq d.fastq e.fastq f.fastq g.fastq h.fastq > bench_proj/badread/exp3/2024/rep1.fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94dd7d86",
   "metadata": {},
   "source": [
    "ASSEMBLY GENERATION: Hybrid assemblies were generated using simulated ART short reads experiment 2 low, replicate 1 to test the effect of the long reads on assembly accuracy. \n",
    "NOTE: Each tool only shows one replicate generation to save space. Assemblies were generated for each tier (2018 or 2024) and for each replicate per tier (1, 2, 3). Only folder location differs in further code."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6660ad98",
   "metadata": {},
   "source": [
    "UNICYCLER:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "718da7fe",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose unicycler -1 bench_proj/ART/exp2low/rep1_1.fq -2 bench_proj/ART/exp2low/rep1_2.fq -l bench_proj/badread/exp3/2018/rep1.fastq -t 32 -o bench_proj/unicycler/exp3/2018/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a5817555",
   "metadata": {},
   "source": [
    "FLYE:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "11381875",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose flye --nano-raw bench_proj/badread/exp3/2018/rep1.fastq --genome-size 4.7m -t 32 --out-dir bench_proj/flye/exp3/2018/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ab01de08",
   "metadata": {},
   "source": [
    "SPADES:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "265888fb",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "/usr/bin/time --verbose spades.py -1 bench_proj/ART/exp2low/rep1_1.fq -2 bench_proj/ART/exp2low/rep1_2.fq --nanopore bench_proj/badread/exp3/2018/rep1.fastq -t 32 -o bench_proj/spades/exp3/2018/rep1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05379a4e",
   "metadata": {},
   "source": [
    "QUAST: Assembly accuracy was measured with QUAST."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd5b389d",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "quast.py flye/exp1high/rep1/flyeE1H1.fasta flye/exp1high/rep2/flyeE1H2.fasta flye/exp1high/rep3/flyeE1H3.fasta flye/exp1mid/rep1/flyeE1M1.fasta flye/exp1mid/rep2/flyeE1M2.fasta flye/exp1mid/rep3/flyeE1M3.fasta flye/exp1low/rep1/flyeE1L1.fasta flye/exp1low/rep2/flyeE1L2.fasta flye/exp1low/rep3/flyeE1L3.fasta flye/exp2high/rep1/flyeE2H1.fasta flye/exp2high/rep2/flyeE2H2.fasta flye/exp2high/rep3/flyeE2H3.fasta flye/exp2mid/rep1/flyeE2M1.fasta flye/exp2mid/rep2/flyeE2M2.fasta flye/exp2mid/rep3/flyeE2M3.fasta flye/exp2low/rep1/flyeE2L1.fasta flye/exp2low/rep2/flyeE2L2.fasta flye/exp2low/rep3/flyeE2L3.fasta unicycler/exp1high/rep1/uniE1H1.fasta unicycler/exp1high/rep2/uniE1H2.fasta unicycler/exp1high/rep3/uniE1H3.fasta unicycler/exp1mid/rep1/uniE1M1.fasta unicycler/exp1mid/rep2/uniE1M2.fasta unicycler/exp1mid/rep3/uniE1M3.fasta unicycler/exp1low/rep1/uniE1L1.fasta unicycler/exp1low/rep2/uniE1L2.fasta unicycler/exp1low/rep3/uniE1L3.fasta unicycler/exp2high/rep1/uniE2H1.fasta unicycler/exp2high/rep2/uniE2H2.fasta unicycler/exp2high/rep3/uniE2H3.fasta unicycler/exp2mid/rep1/uniE2M1.fasta unicycler/exp2mid/rep2/uniE2M2.fasta unicycler/exp2mid/rep3/uniE2M3.fasta unicycler/exp2low/rep1/uniE2L1.fasta unicycler/exp2low/rep2/uniE2L2.fasta unicycler/exp2low/rep3/uniE2L3.fasta spades/exp1high/rep1/spadesE1H1.fasta spades/exp1high/rep1/BRRspadesE1H1.fasta spades/exp1high/rep2/spadesE1H2.fasta spades/exp1high/rep2/BRRspadesE1H2.fasta spades/exp1high/rep3/spadesE1H3.fasta spades/exp1high/rep3/BRRspadesE1H3.fasta spades/exp1mid/rep1/spadesE1M1.fasta spades/exp1mid/rep1/BRRspadesE1M1.fasta spades/exp1mid/rep2/spadesE1M2.fasta spades/exp1mid/rep2/BRRspadesE1M2.fasta spades/exp1mid/rep3/spadesE1M3.fasta spades/exp1mid/rep3/BRRspadesE1M3.fasta spades/exp1low/rep1/spadesE1L1.fasta spades/exp1low/rep1/BRRspadesE1L1.fasta spades/exp1low/rep2/spadesE1L2.fasta spades/exp1low/rep2/BRRspadesE1L2.fasta spades/exp1low/rep3/spadesE1L3.fasta spades/exp1low/rep3/BRRspadesE1L3.fasta spades/exp2high/rep1/spadesE2H1.fasta spades/exp2high/rep1/BRRspadesE2H1.fasta spades/exp2high/rep2/spadesE2H2.fasta spades/exp2high/rep2/BRRspadesE2H2.fasta spades/exp2high/rep3/spadesE2H3.fasta spades/exp2high/rep3/BRRspadesE2H3.fasta spades/exp2mid/rep1/spadesE2M1.fasta spades/exp2mid/rep1/BRRspadesE2M1.fasta spades/exp2mid/rep2/spadesE2M2.fasta spades/exp2mid/rep2/BRRspadesE2M2.fasta spades/exp2mid/rep3/spadesE2M3.fasta spades/exp2mid/rep3/BRRspadesE2M3.fasta spades/exp2low/rep1/spadesE2L1.fasta spades/exp2low/rep1/BRRspadesE2L1.fasta spades/exp2low/rep2/spadesE1L2.fasta spades/exp2low/rep2/BRRspadesE2L2.fasta spades/exp2low/rep3/spadesE2L3.fasta spades/exp2low/rep3/BRRspadesE2L3.fasta flye/exp3/2018/rep1/flyeE3_18_1.fasta flye/exp3/2018/rep2/flyeE3_18_2.fasta flye/exp3/2018/rep3/flyeE3_18_3.fasta flye/exp3/2024/rep1/flyeE3_24_1.fasta flye/exp3/2024/rep2/flyeE3_24_2.fasta flye/exp3/2024/rep3/flyeE3_24_3.fasta unicycler/exp3/2018/rep1/uniE3_18_1.fasta unicycler/exp3/2018/rep2/uniE3_18_2.fasta unicycler/exp3/2018/rep3/uniE3_18_3.fasta unicycler/exp3/2024/rep1/uniE3_24_1.fasta unicycler/exp3/2024/rep2/uniE3_24_2.fasta unicycler/exp3/2024/rep3/uniE3_24_3.fasta spades/exp3/2018/rep1/spadesE3_18_1.fasta spades/exp3/2018/rep1/BRRspadesE3_18_1.fasta spades/exp3/2018/rep2/spadesE3_18_2.fasta spades/exp3/2018/rep2/BRRspadesE3_18_2.fasta spades/exp3/2018/rep3/spadesE3_18_3.fasta spades/exp3/2018/rep3/BRRspadesE3_18_3.fasta spades/exp3/2024/rep1/spadesE3_24_1.fasta spades/exp3/2024/rep1/BRRspadesE3_24_1.fasta spades/exp3/2024/rep2/spadesE3_24_2.fasta spades/exp3/2024/rep2/BRRspadesE3_24_1.fasta spades/exp3/2024/rep3/spadesE3_24_3.fasta spades/exp3/2024/rep3/BRRspadesE3_24_1.fasta -r BL21.fasta -o QUAST/allofit"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9458b2cd",
   "metadata": {},
   "source": [
    "BLAST: 16S Assembly accuracy was measured with BLAST."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "27270ae9",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "for i in flye/exp1high/rep1/flyeE1H1.fasta flye/exp1high/rep2/flyeE1H2.fasta flye/exp1high/rep3/flyeE1H3.fasta flye/exp1mid/rep1/flyeE1M1.fasta flye/exp1mid/rep2/flyeE1M2.fasta flye/exp1mid/rep3/flyeE1M3.fasta flye/exp1low/rep1/flyeE1L1.fasta flye/exp1low/rep2/flyeE1L2.fasta flye/exp1low/rep3/flyeE1L3.fasta flye/exp2high/rep1/flyeE2H1.fasta flye/exp2high/rep2/flyeE2H2.fasta flye/exp2high/rep3/flyeE2H3.fasta flye/exp2mid/rep1/flyeE2M1.fasta flye/exp2mid/rep2/flyeE2M2.fasta flye/exp2mid/rep3/flyeE2M3.fasta flye/exp2low/rep1/flyeE2L1.fasta flye/exp2low/rep2/flyeE2L2.fasta flye/exp2low/rep3/flyeE2L3.fasta unicycler/exp1high/rep1/uniE1H1.fasta unicycler/exp1high/rep2/uniE1H2.fasta unicycler/exp1high/rep3/uniE1H3.fasta unicycler/exp1mid/rep1/uniE1M1.fasta unicycler/exp1mid/rep2/uniE1M2.fasta unicycler/exp1mid/rep3/uniE1M3.fasta unicycler/exp1low/rep1/uniE1L1.fasta unicycler/exp1low/rep2/uniE1L2.fasta unicycler/exp1low/rep3/uniE1L3.fasta unicycler/exp2high/rep1/uniE2H1.fasta unicycler/exp2high/rep2/uniE2H2.fasta unicycler/exp2high/rep3/uniE2H3.fasta unicycler/exp2mid/rep1/uniE2M1.fasta unicycler/exp2mid/rep2/uniE2M2.fasta unicycler/exp2mid/rep3/uniE2M3.fasta unicycler/exp2low/rep1/uniE2L1.fasta unicycler/exp2low/rep2/uniE2L2.fasta unicycler/exp2low/rep3/uniE2L3.fasta spades/exp1high/rep1/spadesE1H1.fasta spades/exp1high/rep1/BRRspadesE1H1.fasta spades/exp1high/rep2/spadesE1H2.fasta spades/exp1high/rep2/BRRspadesE1H2.fasta spades/exp1high/rep3/spadesE1H3.fasta spades/exp1high/rep3/BRRspadesE1H3.fasta spades/exp1mid/rep1/spadesE1M1.fasta spades/exp1mid/rep1/BRRspadesE1M1.fasta spades/exp1mid/rep2/spadesE1M2.fasta spades/exp1mid/rep2/BRRspadesE1M2.fasta spades/exp1mid/rep3/spadesE1M3.fasta spades/exp1mid/rep3/BRRspadesE1M3.fasta spades/exp1low/rep1/spadesE1L1.fasta spades/exp1low/rep1/BRRspadesE1L1.fasta spades/exp1low/rep2/spadesE1L2.fasta spades/exp1low/rep2/BRRspadesE1L2.fasta spades/exp1low/rep3/spadesE1L3.fasta spades/exp1low/rep3/BRRspadesE1L3.fasta spades/exp2high/rep1/spadesE2H1.fasta spades/exp2high/rep1/BRRspadesE2H1.fasta spades/exp2high/rep2/spadesE2H2.fasta spades/exp2high/rep2/BRRspadesE2H2.fasta spades/exp2high/rep3/spadesE2H3.fasta spades/exp2high/rep3/BRRspadesE2H3.fasta spades/exp2mid/rep1/spadesE2M1.fasta spades/exp2mid/rep1/BRRspadesE2M1.fasta spades/exp2mid/rep2/spadesE2M2.fasta spades/exp2mid/rep2/BRRspadesE2M2.fasta spades/exp2mid/rep3/spadesE2M3.fasta spades/exp2mid/rep3/BRRspadesE2M3.fasta spades/exp2low/rep1/spadesE2L1.fasta spades/exp2low/rep1/BRRspadesE2L1.fasta spades/exp2low/rep2/spadesE2L2.fasta spades/exp2low/rep2/BRRspadesE2L2.fasta spades/exp2low/rep3/spadesE2L3.fasta spades/exp2low/rep3/BRRspadesE2L3.fasta flye/exp3/2018/rep1/flyeE3_18_1.fasta flye/exp3/2018/rep2/flyeE3_18_2.fasta flye/exp3/2018/rep3/flyeE3_18_3.fasta flye/exp3/2024/rep1/flyeE3_24_1.fasta flye/exp3/2024/rep2/flyeE3_24_2.fasta flye/exp3/2024/rep3/flyeE3_24_3.fasta unicycler/exp3/2018/rep1/uniE3_18_1.fasta unicycler/exp3/2018/rep2/uniE3_18_2.fasta unicycler/exp3/2018/rep3/uniE3_18_3.fasta unicycler/exp3/2024/rep1/uniE3_24_1.fasta unicycler/exp3/2024/rep2/uniE3_24_2.fasta unicycler/exp3/2024/rep3/uniE3_24_3.fasta spades/exp3/2018/rep1/spadesE3_18_1.fasta spades/exp3/2018/rep1/BRRspadesE3_18_1.fasta spades/exp3/2018/rep2/spadesE3_18_2.fasta spades/exp3/2018/rep2/BRRspadesE3_18_2.fasta spades/exp3/2018/rep3/spadesE3_18_3.fasta spades/exp3/2018/rep3/BRRspadesE3_18_3.fasta spades/exp3/2024/rep1/spadesE3_24_1.fastaspades/exp3/2024/rep1/BRRspadesE3_24_1.fasta spades/exp3/2024/rep2/spadesE3_24_2.fasta spades/exp3/2024/rep2/BRRspadesE3_24_1.fasta spades/exp3/2024/rep3/spadesE3_24_3.fasta spades/exp3/2024/rep3/BRRspadesE3_24_1.fasta; do     prefix=$(basename \"$i\" .fasta);     sed \"s/^>/>${prefix}_/\" \"$i\" >> expB.fasta; done"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9d7565d0",
   "metadata": {},
   "source": [
    "Generate a concatenated fasta file where each assembly is denoted with its respective file name as the header."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdcef089",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "makeblastdb -in expB.fasta -dbtype nucl -out assembly3B"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8da9c4f0",
   "metadata": {},
   "source": [
    "Make a BLAST database using this fasta file before executing the blast "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dd73e437",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "blastn -query BL21_16S.fasta -db assemblydb -outfmt \"6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore\" -out blast3_simulated.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3c02048b",
   "metadata": {},
   "source": [
    "Call BLAST command query to generate metrics on the 16S region of E. coli BL21 using its reference 16S region. Tried to make each column header its respective name but did not work."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62a1e8be",
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [],
   "source": [
    "tr '\\t' ',' < blast3_simulated.txt > blast3_simulated.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e541b9fb",
   "metadata": {},
   "source": [
    "Edited column headers manuallyin Excel after converting .txt to .csv"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
